grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -259,6 +259,18 @@ public T superType(T superType) {
         return getThis();
     }
 
+    /**
+     * Adds another subtype to this type
+     *
+     * @param type The sub type of this type
+     * @return The Type itself
+     */
+    public T subType(T type){
+        ((TypeImpl) type).putEdge(this, Schema.EdgeLabel.SUB);
+        type(); //Check if there is a circular sub loop
+        return getThis();
+    }
```

Comment:
I'm not sure your circular sub loop check will work here (since it is backwards).
If you define this method in terms of `superType` (so `type.superType(this)`), it will resolve the issue, plus it will be more obvious that this method is just an inverse.